### PR TITLE
Fix undefined behavior found by static code analyzer.

### PR DESCRIPTION
**Pull request by 27thLiz**
Adresses the issue mentioned in https://software.intel.com/en-us/articles/the-ultimate-question-of-programming-refactoring-and-everything

### CODE REVIEW

**Comment by Hinsbart** [comment]
Thanks for fixing our bugs 😄

**Comment by Zylann** [comment]
Did we looked at this? https://www.viva64.com/en/b/0321/

File: scene/animation/tween_interpolaters.cpp [hunk 1]
```diff
@@ -262,7 +262,8 @@ namespace cubic {
 
 	static real_t out(real_t t, real_t b, real_t c, real_t d)
 	{
-		return c * ((t = t / d - 1) * t * t + 1) + b;
+		t = t / d - 1;
```
Comment (akien-mga) [inline scene/animation/tween_interpolaters.cpp]: Might be worth splitting all the ugly inline assignments of this class no?

Comment (Hinsbart) [inline scene/animation/tween_interpolaters.cpp]: Probably, yeah.
But I don't have the time for that sadly - so: "PR welcome" ^^

**Pull request merged**